diff --git a/src/modules/video_capture/linux/device_info_linux.cc b/src/modules/video_capture/linux/device_info_linux.cc
index 9ced0ae..a8ded93 100644
--- a/src/modules/video_capture/linux/device_info_linux.cc
+++ b/src/modules/video_capture/linux/device_info_linux.cc
@@ -211,11 +211,11 @@ int32_t DeviceInfoLinux::FillCapabilities(int fd) {
   unsigned int videoFormats[] = {V4L2_PIX_FMT_MJPEG, V4L2_PIX_FMT_YUV420,
                                  V4L2_PIX_FMT_YUYV, V4L2_PIX_FMT_UYVY};
 
-  int sizes = 13;
+  int sizes = 14;
   unsigned int size[][2] = {{128, 96},   {160, 120},  {176, 144},  {320, 240},
                             {352, 288},  {640, 480},  {704, 576},  {800, 600},
                             {960, 720},  {1280, 720}, {1024, 768}, {1440, 1080},
-                            {1920, 1080}};
+                            {1920, 1080}, {3840, 2160}};
 
   int index = 0;
   for (int fmts = 0; fmts < totalFmts; fmts++) {
diff --git a/src/pc/videocapturertracksource.cc b/src/pc/videocapturertracksource.cc
index 0767dcf..518557b 100644
--- a/src/pc/videocapturertracksource.cc
+++ b/src/pc/videocapturertracksource.cc
@@ -34,6 +34,7 @@ static const cricket::VideoFormatPod kDefaultFormat = {
 
 // List of formats used if the camera doesn't support capability enumeration.
 static const cricket::VideoFormatPod kVideoFormats[] = {
+    {3840, 2160, FPS_TO_INTERVAL(30), cricket::FOURCC_ANY},
     {1920, 1080, FPS_TO_INTERVAL(30), cricket::FOURCC_ANY},
     {1280, 720, FPS_TO_INTERVAL(30), cricket::FOURCC_ANY},
     {960, 720, FPS_TO_INTERVAL(30), cricket::FOURCC_ANY},
