diff --git a/src/media/engine/webrtc_video_engine.cc b/src/media/engine/webrtc_video_engine.cc
index 4219c43..b7b4ffa 100644
--- a/src/media/engine/webrtc_video_engine.cc
+++ b/src/media/engine/webrtc_video_engine.cc
@@ -244,8 +244,10 @@ static int GetMaxDefaultVideoBitrateKbps(int width,
     max_bitrate = 1700;
   } else if (width * height <= 960 * 540) {
     max_bitrate = 2000;
-  } else {
+  } else if (width * height <= 1920 * 1080) {
     max_bitrate = 2500;
+  } else {
+    max_bitrate = 15000;
   }
   if (is_screenshare)
     max_bitrate = std::max(max_bitrate, 1200);
diff --git a/src/modules/video_capture/linux/device_info_linux.cc b/src/modules/video_capture/linux/device_info_linux.cc
index 39f72b0..3cb1226 100644
--- a/src/modules/video_capture/linux/device_info_linux.cc
+++ b/src/modules/video_capture/linux/device_info_linux.cc
@@ -224,11 +224,11 @@ int32_t DeviceInfoLinux::FillCapabilities(int fd) {
   unsigned int videoFormats[] = {V4L2_PIX_FMT_MJPEG, V4L2_PIX_FMT_YUV420,
                                  V4L2_PIX_FMT_YUYV, V4L2_PIX_FMT_UYVY};
 
-  int sizes = 13;
+  int sizes = 14;
   unsigned int size[][2] = {{128, 96},   {160, 120},  {176, 144},  {320, 240},
                             {352, 288},  {640, 480},  {704, 576},  {800, 600},
                             {960, 720},  {1280, 720}, {1024, 768}, {1440, 1080},
-                            {1920, 1080}};
+                            {1920, 1080}, {3840, 2160}};
 
   for (int fmts = 0; fmts < totalFmts; fmts++) {
     for (int i = 0; i < sizes; i++) {
