diff --git a/src/civetweb.c b/src/civetweb.c
index c0ccbaa2..b12ba0c8 100644
--- a/src/civetweb.c
+++ b/src/civetweb.c
@@ -16461,6 +16461,18 @@ master_thread_run(void *thread_func_param)
 #endif
 	(void)pthread_mutex_unlock(&ctx->thread_mutex);
 
+	/* send signal to awake up worker on consume_socket */
+    mg_sleep(100);
+	(void)pthread_mutex_lock(&ctx->thread_mutex);
+#if defined(ALTERNATIVE_QUEUE)
+	for (i = 0; i < ctx->cfg_worker_threads; i++) {
+		event_signal(ctx->client_wait_events[i]);
+	}
+#else
+	pthread_cond_broadcast(&ctx->sq_full);
+#endif
+	(void)pthread_mutex_unlock(&ctx->thread_mutex);
+
 	/* Join all worker threads to avoid leaking threads. */
 	workerthreadcount = ctx->cfg_worker_threads;
 	for (i = 0; i < workerthreadcount; i++) {
