diff --git a/src/sdk/objc/components/video_codec/RTCVideoEncoderH264.mm b/src/sdk/objc/components/video_codec/RTCVideoEncoderH264.mm
index 4028e7acee..585743e1aa 100644
--- a/src/sdk/objc/components/video_codec/RTCVideoEncoderH264.mm
+++ b/src/sdk/objc/components/video_codec/RTCVideoEncoderH264.mm
@@ -554,6 +554,7 @@ NSUInteger GetMaxSampleRate(const webrtc::H264::ProfileLevelId &profile_level_id
   OSType framePixelFormat = [self pixelFormatOfFrame:frame];
 
   if (_compressionSession) {
+    _pixelBufferPool = VTCompressionSessionGetPixelBufferPool(_compressionSession);
     // The pool attribute `kCVPixelBufferPixelFormatTypeKey` can contain either an array of pixel
     // formats or a single pixel format.
     NSDictionary *poolAttributes =