diff --git a/bthird_party/oringssl/BUILD.gn b/third_party/boringssl/BUILD.gn
index a8aa9c9ece4..9e24ca7d8a9 100644
--- a/third_party/boringssl/BUILD.gn
+++ b/third_party/boringssl/BUILD.gn
@@ -86,7 +86,11 @@ if (is_win && !is_msan && current_cpu != "arm64") {
       }
     } else if (current_cpu == "arm") {
       if (is_linux || is_android) {
-        sources += crypto_sources_linux_arm
+        if (arm_version <= 6) {
+          public_configs = [ ":no_asm_config" ]
+        } else {
+          sources += crypto_sources_linux_arm
+        }
       } else if (is_ios) {
         sources += crypto_sources_ios_arm
       } else {
diff --git a/third_party/opus/BUILD.gn b/third_party/opus/BUILD.gn
index b54f8568823..a5c7d197f4d 100644
--- a/third_party/opus/BUILD.gn
+++ b/third_party/opus/BUILD.gn
@@ -21,7 +21,7 @@ use_opus_x86_optimization =
 # If OPUS Run Time CPU Detections (RTCD) shall be used.
 # Based on the conditions in celt/arm/armcpu.c:
 # defined(_MSC_VER) || defined(__linux__).
-use_opus_arm_rtcd = current_cpu == "arm" && (is_win || is_android || is_linux)
+use_opus_arm_rtcd = current_cpu == "arm" && (is_win || is_android || is_linux) && (arm_version >= 7)
 
 config("opus_config") {
   include_dirs = [ "src/include" ]
